### ISSUE TITLE
Error: could not stabilize within 2m0s: context deadline exceeded

### ISSUE THREAD

**Issue by james-freitas** [GoogleCloudPlatform/cloud-code-intellij#2749]
Hi, here are the version of the tools I am using:
- Cloud Code for IntelliJ version: 20.4.2
- Google Cloud SDK version: 289.0.0
- About my OS
```bash
lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.4 LTS
Release:	18.04
Codename:	bionic
```

**What did you do?**
I am following this tutorial:  https://www.infoq.com/articles/spring-boot-tutorial/?utm_campaign=infoq_content&utm_source=infoq&utm_medium=feed&utm_term=Java

I am in the step when I click on the **Develop on Kubernets** play button.  

**What did you expect to see?**
My expectation according to the tutorial: 
> Cloud Code will build the app, create the image, deploy the application to your GKE cluster and stream the logs from Stackdriver into your local machine.

**What did you see instead?**
When I do that I get this error:
```bash
- deployment/dog-service failed. Error: could not stabilize within 2m0s: context deadline exceeded.
```
Here is the full log of the error:
```bash
/home/james/.cache/cloud-code/bin/versions/8d4ac68282769dd0424264593410c5cdd6959903d62db03edb1a62c2d1e5b4da/skaffold dev --filename skaffold.yaml --label ide=Idea --label ideVersion=2020.1.0.0 --label ijPluginVersion=unknown --rpc-port 50051 --port-forward=true --status-check=true --enable-rpc=true
Listing files to watch...
 - gcr.io/**my-project**/dog
Generating tags...
 - gcr.io/**my-project**/dog -> gcr.io/**my-project**/dog:dirty
Checking cache...
 - gcr.io/**my-project**/dog: Found Locally
time="2020-04-21T11:09:25-03:00" level=warning msg="Unable to find git commit: running [git describe --tags --always]\n - stdout: \n - stderr: \"fatal: not a git repository (or any parent up to mount point /)\\nStopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).\\n\": exit status 128"
Tags used in deployment:
 - gcr.io/**my-project**/dog -> gcr.io/**my-project**/dog:7f72e5508a9b74f2b775966f2075efa60ec7336f31f0b0ae17ff153b8130ef5d
   local images can't be referenced by digest. They are tagged and referenced by a unique ID instead
Starting deploy...
 - deployment.apps/dog-service created
 - service/dog-service created
Waiting for deployments to stabilize...
 - deployment/dog-service: waiting for rollout to finish: 0 of 1 updated replicas are available...
 - deployment/dog-service failed. Error: could not stabilize within 2m0s: context deadline exceeded.
Cleaning up...
 - deployment.apps "dog-service" deleted
 - service "dog-service" deleted
time="2020-04-21T11:11:26-03:00" level=fatal msg="exiting dev mode because first deploy failed: 1/1 deployment(s) failed"
```
I don't understand what I am doing wrong here.  Could you take a look?

Thanks in advance,
James

**Comment by etanshaul** [comment]
hi @james-freitas 

This could be a number of things (and we are working on improving error messaging). 

One thing to check is after you run the session (before you terminate it), to `describe` the pod and share the output (I'm looking for any error messages in the pod state). You could either do:
- `kubectle describe [name_of_pod]`
- or, use the Kubernetes Explorer built into the Cloud Code, drill down into the pod within your cluster, right-click -> describe.

**Comment by james-freitas** [comment]
hi @etanshaul 

I am newbie in Kubernetes so I don't understand much :smile:  In the picture below in the Kubernetes Explorer is this one I should right-click > describe?

![image](https://user-images.githubusercontent.com/4712453/79880745-cb8cff80-83c6-11ea-8fb3-7c160516bc58.png)

**Comment by etanshaul** [comment]
no problem!

So that is the node that you described, I was looking for the `pod` output:

Under `minikube > Namespaces > default > Workloads > Pods` you _should_ see a pod there after running the deployment from cloud code. If you right click on that pod and select `Describe` you should see the output.

Note: if you don't see any running pods, then ensure that you leave the Cloud Code Kubernetes deployment session running. Once you hit the stop button, Cloud Code (by default) will clean up the resources and terminate the pods / services.

**Comment by james-freitas** [comment]
Thanks!

As you said:

![image](https://user-images.githubusercontent.com/4712453/79881877-283cea00-83c8-11ea-8e85-1b82078253fb.png)

Here is the output:
```bash
/home/james/.cache/cloud-code/bin/versions/8d4ac68282769dd0424264593410c5cdd6959903d62db03edb1a62c2d1e5b4da/kubectl describe pods dog-service-86784f5cf4-v7w5p --namespace default
Name:         dog-service-86784f5cf4-v7w5p
Namespace:    default
Priority:     0
Node:         minikube/192.168.99.100
Start Time:   Tue, 21 Apr 2020 12:02:25 -0300
Labels:       app=dog-service
              app.kubernetes.io/managed-by=skaffold-v1.7.0
              ide=Idea
              ideVersion=2020.1.0.0
              ijPluginVersion=unknown
              pod-template-hash=86784f5cf4
              skaffold.dev/builder=local
              skaffold.dev/cleanup=true
              skaffold.dev/deployer=kubectl
              skaffold.dev/docker-api-version=1.40
              skaffold.dev/run-id=dd5970fb-241d-4a87-88ed-c0d7847e6860
              skaffold.dev/tag-policy=git-commit
              skaffold.dev/tail=true
Annotations:  <none>
Status:       Running
IP:           172.17.0.4
IPs:
  IP:           172.17.0.4
Controlled By:  ReplicaSet/dog-service-86784f5cf4
Containers:
  dog-service:
    Container ID:   docker://4ec53e6c7a2a6609a160e98a024896f7bbdcbb274f1c7e1a34be85f435c8e8b8
    Image:          gcr.io/spring-project-273610/dog:7f72e5508a9b74f2b775966f2075efa60ec7336f31f0b0ae17ff153b8130ef5d
    Image ID:       docker://sha256:7f72e5508a9b74f2b775966f2075efa60ec7336f31f0b0ae17ff153b8130ef5d
    Port:           8080/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Tue, 21 Apr 2020 12:02:51 -0300
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Tue, 21 Apr 2020 12:02:32 -0300
      Finished:     Tue, 21 Apr 2020 12:02:37 -0300
    Ready:          False
    Restart Count:  2
    Liveness:       http-get http://:8080/actuator/health delay=20s timeout=1s period=10s #success=1 #failure=3
    Readiness:      http-get http://:8080/actuator/health delay=30s timeout=1s period=10s #success=1 #failure=3
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-8k4ms (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  default-token-8k4ms:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-8k4ms
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason     Age                From               Message
  ----     ------     ----               ----               -------
  Normal   Scheduled  <unknown>          default-scheduler  Successfully assigned default/dog-service-86784f5cf4-v7w5p to minikube
  Warning  BackOff    15s (x2 over 17s)  kubelet, minikube  Back-off restarting failed container
  Normal   Pulled     4s (x3 over 29s)   kubelet, minikube  Container image "gcr.io/spring-project-273610/dog:7f72e5508a9b74f2b775966f2075efa60ec7336f31f0b0ae17ff153b8130ef5d" already present on machine
  Normal   Created    4s (x3 over 29s)   kubelet, minikube  Created container dog-service
  Normal   Started    4s (x3 over 29s)   kubelet, minikube  Started container dog-service
```

**Comment by etanshaul** [comment]
it looks like the pod was successfully scheduled. any you are still seeing `Error: could not stabilize within 2m0s: context deadline exceeded` in the log right? We'll try and reproduce this.

**Comment by etanshaul** [comment]
this issue looks very similar to this one: https://github.com/GoogleCloudPlatform/cloud-code-samples/issues/302

Although the non-normal deployment status may suggest otherwise, I'm wondering if you can check to see if the app still started up normally despite that message about failing to stabilize:

After running the deployment, in the IDE's event log you should see some links to the locally port-forwarded app printed. E.g:

<img width="1124" alt="Screen Shot 2020-04-21 at 1 33 00 PM" src="https://user-images.githubusercontent.com/1735744/79895280-a4d5c580-83d4-11ea-9d6c-6307aa975ab5.png">

If you click on that `localhost` link, you should be able to see if the app is running or not.

**Comment by james-freitas** [comment]
Here is my **Event Log**:

![image](https://user-images.githubusercontent.com/4712453/79897120-9f31ad00-83df-11ea-94be-6301972855d5.png)

Unfortunatelly is different because there is no port forwarding :cry:

**Comment by etanshaul** [comment]
hi @james-freitas - apologies for the delayed reply. Are you still running into this issue? 

We've released Cloud Code version `20.5.1` which points to the latest version of Skaffold which may or may not help here. 

If you still have the issue, could you please paste again the entire session log output with DEBUG verbosity (to configure DEBUG verbosity it is one of the last settings inside of the run configuration):

![image](https://user-images.githubusercontent.com/1735744/83269351-f8d28780-a194-11ea-9f27-185478559baa.png)

**Comment by jmah-cc** [comment]
@james-freitas Closing since we believe this has been fixed. Please re-open if you see this issue again.
